pyocd/probe/stlink/stlink.py

```diff
@@ -63,6 +63,8 @@ def __init__(self, device):
     
     def open(self):
         self._device.open()
+        if (self.get_current_mode() == Commands.DEV_DFU_MODE):
+            self.exit_dfu_mode()
```

Comment:
There is already an `enter_idle()` method that can exit DFU mode. I think it would work if you called `enter_idle()` here.

Reply:
Updated the pull-request, works for me!